packages/graphql/src/lib.ts

```diff
@@ -1,8 +1,102 @@
+   * Output file will interpolate the following values:
+   * - schema-name: Name of the schema if multiple
+   *
+   * @default `{schema-name}.graphql`
+   *
+   * @example Single schema
+   * - `schema.graphql`
+   *
+   * @example Multiple schemas
+   * - `Org1.Schema1.graphql`
+   * - `Org1.Schema2.graphql`
+   */
+  "output-file"?: string;
+
+  /**
+   * Set the newline character for emitting files.
+   * @default lf
+   */
+  "new-line"?: "crlf" | "lf";
+
+  /**
+   * Omit unreachable types.
+   * By default all types declared under the schema namespace will be included. With this flag on only types references in an operation will be emitted.
+   * @default false
+   */
+  "omit-unreachable-types"?: boolean;
+
+  /**
+   * Only emit types if a correct GraphQL translation type is found. Don't emit Any types and operations that don't have the GraphQL decorators.
+   * By default a best effort is made to emit all types.
+   * @default false
+   */
+  "strict-emit"?: boolean;
+}
+
+const EmitterOptionsSchema: JSONSchemaType<GraphQLEmitterOptions> = {
+  type: "object",
+  additionalProperties: false,
+  properties: {
+    "output-file": {
+      type: "string",
+      nullable: true,
+      description: [
+        "Name of the output file.",
+        " Output file will interpolate the following values:",
+        " - schema-name: Name of the schema if multiple",
+        "",
+        " Default: `{schema-name}.graphql`",
+        "",
+        " Example Single schema",
+        " - `schema.graphql`",
+        "",
+        " Example Multiple schemas",
+        " - `Org1.Schema1.graphql`",
+        " - `Org1.Schema2.graphql`",
+      ].join("\n"),
+    },
+    "new-line": {
+      type: "string",
+      enum: ["crlf", "lf"],
+      default: "lf",
+      nullable: true,
+      description: "Set the newLine character for emitting files.",
+    },
+    "omit-unreachable-types": {
+      type: "boolean",
+      nullable: true,
+      description: [
+        "Omit unreachable types.",
+        "By default all types declared under the schema namespace will be included.",
+        "With this flag on only types references in an operation will be emitted.",
+      ].join("\n"),
+    },
+    "strict-emit": {
+      type: "boolean",
+      nullable: true,
+      description: [
+        "Only emit types if a correct GraphQL translation type is found.",
+        "Don't emit Any types and operations that don't have the GraphQL decorators.",
+        "By default a best effort is made to emit all types.",
+      ].join("\n"),
+    },
+  },
+  required: [],
+};
+
+export const libDef = {
   name: "@typespec/graphql",
   diagnostics: {},
-});
+  emitter: {
+    options: EmitterOptionsSchema as JSONSchemaType<GraphQLEmitterOptions>,
+  },
+} as const;
+
+export const $lib = createTypeSpecLibrary(libDef);
 
-export const { reportDiagnostic, createDiagnostic } = $lib;
+export const { reportDiagnostic, createDiagnostic, createStateSymbol } = $lib;
```

Comment:
For state I recommend you use the `state:` property of when creating the lib to define all your state keys instead of this function.

Reply:
combined with some of the currently experimental `unsafe_useStateMap, unsafe_useStateSet` it makes it much easier to do simple data accessor decorators(check paging or streaming decorators for examples)

Reply:
Sounds good, that matches with the way we're implemented it on the prototype (I think OAI does the same). Will remove.

Reply:
done